src/installer/corehost/cli/bundle/extractor.cpp

```diff
@@ -35,7 +35,14 @@ pal::string_t& extractor_t::extraction_dir()
         {
             pal::string_t current_dir = _X(".");
             pal::string_t relative_path(m_extraction_dir);
-            m_extraction_dir = pal::realpath(&current_dir);
+            if (!pal::realpath(&current_dir))
+            {
+                trace::error(_X("Failure processing application bundle."));
+                trace::error(_X("Failed to obtain the current directory for extracting bundled files to relative path [%s]."), m_extraction_dir.c_str());
+                throw StatusCode::BundleExtractionFailure;
```

Comment:
We should probably add at least one negative test case - passing nonsense to the input and make this fail.